go_backend/main.go

```diff
@@ -0,0 +1,242 @@
+package main
+
+import (
+	"database/sql"
+	"fmt"
+	"log"
+
+	"github.com/gin-gonic/gin"
+	_ "github.com/go-sql-driver/mysql"
+)
+
+func homePage(c *gin.Context) {
+	c.JSON(200, gin.H{
+		"message": "Hello World",
+	})
+}
+
+// func postHomePage(c *gin.Context) {
+// 	body := c.Request.Body
+// 	value, err := ioutil.ReadAll(body)
+// 	if err != nil {
+// 		fmt.Println(err.Error())
+// 	}
+// 	c.JSON(200, gin.H{
+// 		"message": string(value),
+// 	})
+// }
+
+//sign up function
+func querySignUp(c *gin.Context) {
+	//getting data from request
+	// userID := c.Query("userID")
+	email := c.Query("email")
+	password := c.Query("password")
+	firstName := c.Query("firstName")
+	lastName := c.Query("lastName")
+	phoneNumber := c.Query("phoneNumber")
+	userIdentity := c.Query("userIdentity")
+
+	//connect to the db
+	db, err := sql.Open("mysql", "root:@tcp(127.0.0.1:3306)/foodapp")
+	if err != nil {
+		fmt.Println("DB connection error")
+		c.JSON(200, gin.H{
+			"httpCode": "500",
+			"message":  "Database is not connected",
+		})
+		panic(err.Error())
+	} else {
+		fmt.Println("DB Connected!")
+	}
+	defer db.Close()
+
+	//insert to db
+	insert, err := db.Query("INSERT INTO userinfo VALUES(DEFAULT,?,?,?,?,?,?)", email, password, firstName, lastName, phoneNumber, userIdentity)
+
+	if err != nil {
+		fmt.Println("Sign up error")
+		c.JSON(200, gin.H{
+			"httpCode": "500",
+			"message":  "Signup is not vailed",
+		})
+		panic(err.Error())
+	} else {
+		c.JSON(200, gin.H{
+			"httpCode": "201",
+			"message":  "User created",
+		})
+	}
+	defer insert.Close()
+}
+
+//login function
+func queryLogin(c *gin.Context) {
+	//getting data from request
+	email := c.Query("email")
+	password := c.Query("password")
+
+	//connect to the db
+	db, err := sql.Open("mysql", "root:@tcp(127.0.0.1:3306)/foodapp")
+	if err != nil {
+		fmt.Println("DB connection error")
+		c.JSON(200, gin.H{
+			"httpCode": "500",
+			"message":  "Database is not connected",
+		})
+		panic(err.Error())
+	} else {
+		fmt.Println("DB Connected!")
+	}
+	defer db.Close()
+
+	//select userID from db
+	var (
+		userID int
+	)
+	rows, err := db.Query("SELECT userID from userinfo WHERE email = ? AND password = ?", email, password)
+	if err != nil {
+		fmt.Println("Login error")
+		c.JSON(200, gin.H{
+			"httpCode": "404",
+			"message":  "Such username or password is incorrect",
+		})
+		panic(err.Error())
+	} else {
+		for rows.Next() {
+			err := rows.Scan(&userID)
+			if err != nil {
+				log.Fatal(err)
+			}
+		}
+		if userID == 0 {
```

Comment:
Why userID will equal to 0?

Reply:
userID is an auto increment variable, which starts from 1. If the user is not in the list, a default userID will be 0 as what it returns, so if I see the userID returned is 0, then it means no such as a user